### ISSUE TITLE
Ftruncate failed when the original file size = 0 and the extended part is not continuous clusters.

### ISSUE THREAD

**Issue by Erichorng** [namjaejeon/linux-exfat-oot#66]
I built the exfat module with the latest code on mainline.
I encountered a bug during the bellow experiment:

steps to reproduce:
1. create 4G exfat filesystem.
2. create a file and truncate to 3G.
3. delete the file.
4. repeat step 2 again.
5. receive EIO, ftruncate failed.

Here is the script I used for testing:

```
#include <iostream>
#include <fcntl.h>
#include <unistd.h>

int main() {
    const char* filename = "example_file";
    off_t file_size = 3LL * 1024 * 1024 * 1024;

    // Open the file or create it if it doesn't exist
    int fd = open(filename, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
    if (fd == -1) {
        std::cerr << "Error opening/creating the file" << std::endl;
        return 1;
    }

    // Set the file size using ftruncate
    if (ftruncate(fd, file_size) == -1) {
        std::cerr << "Error setting file size using ftruncate" << std::endl;
        close(fd);
        return 1;
    }

    // Cleanup
    close(fd);

    return 0;
}
```

and this is a patch I wrote to fix the bug:
[0001-fix-bug-ftruncate-failed.patch](https://github.com/namjaejeon/linux-exfat-oot/files/14189263/0001-fix-bug-ftruncate-failed.patch)

**Comment by namjaejeon** [comment]
> create 4G exfat filesystem.

This issue is not happening on large exfat partition ?

**Comment by Erichorng** [comment]
By the way, I'd like to share the results of the experiment. If it's not a 4GB file system but a larger one, let's say 8GB, and you repeatedly create and delete 1GB files, you'll encounter issues on the eighth creation. The reason is that during the eighth ftruncate, the expanded portion consists of non-contiguous clusters.

**Comment by namjaejeon** [comment]
@Erichorng Right. Thanks for your check:) @YuezhangMo will hopefully check this issue next week.

**Comment by teletypo** [comment]
Hi @namjaejeon and @YuezhangMo! Is this bug still present in latest mainline linux? If so, any chance you can merge Erich's patch?

**Comment by namjaejeon** [comment]
@teletypo Problem is happening with the latest kernel ?
